### PR TITLE
Asnide 188 placeholder for import from component library wizard

### PR DESCRIPTION
**Pull request by psockoN7**

### CODE REVIEW

File: asn1acn.pro [hunk 1]
```diff
@@ -128,6 +128,8 @@ SOURCES += \
     libraries/modulemetadataparser.cpp \
     libraries/generalmetadataparser.cpp \
     \
+    component-importer\importcomponentdialog.cpp \
```
Comment (hcorg) [inline asn1acn.pro]: I wonder should it have it's own folder or be part of `libraries` folder (or have some subdir there)

File: src/asn1acn.cpp [hunk 2]
```diff
@@ -174,6 +177,16 @@ void Asn1AcnPlugin::initializeMenus()
...
 
+
+    asnToolsMenu->addSeparator();
+    QAction *importFromAsnComponents = new QAction(tr("Import from ASN.1 components library..."), this);
+    command = Core::ActionManager::registerAction(importFromAsnComponents, Constants::IMPORT_FROM_COMPONENTS_LIBRARY);
+    connect(importFromAsnComponents, &QAction::triggered, this, &Asn1AcnPlugin::raiseImportComponentWindow);
+    asnToolsMenu->addAction(command);
+    importFromAsnComponents->setEnabled(ProjectExplorer::SessionManager::hasProjects());
+    connect(ProjectExplorer::ProjectExplorerPlugin::instance(), &ProjectExplorer::ProjectExplorerPlugin::fileListChanged,
+            [importFromAsnComponents]() { importFromAsnComponents->setEnabled(ProjectExplorer::SessionManager::hasProjects()); });
```
Comment (hcorg) [inline src/asn1acn.cpp]: I think it's a good moment to do some refactoring in this method - it got awfully long, we could at least split it some like "createImportAsnComponentsAction" etc.

File: src/component-importer/importcomponentdialog.cpp [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+ImportComponentDialog::ImportComponentDialog(QWidget *parent)
+    : QDialog(parent)
+    , m_libraries(Settings::load<Settings::Libraries>())
+{
+    m_ui.setupUi(this);
+
+    setAttribute(Qt::WA_DeleteOnClose);
+    connect(Core::ICore::instance(), &Core::ICore::coreAboutToClose, this, &QWidget::close);
+
+    connect(m_libraries.get(), &Settings::Libraries::changed, this, &ImportComponentDialog::refreshPaths);
```
Comment (psockoN7) [inline src/component-importer/importcomponentdialog.cpp]: I feel obliged to mention that this connect is useless now, as changed signal is not emitted anywhere. I did not wanted to spend time on fixing it as this part may, or may not be used later. In final implementation it should be done though.

File: src/component-importer/importcomponentdialog.cpp [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+    else
+        m_ui.builtInCombo->clear();
+}
+
+void ImportComponentDialog::accept()
+{
+    if (m_ui.builtInRadio->isChecked())
+        addFilesFromDirectory(m_ui.builtInCombo->currentText());
+    else if (m_ui.customRadio->isChecked())
+        addFilesFromDirectory(m_ui.asn1sccPathChooser->path());
```
Comment (hcorg) [inline src/component-importer/importcomponentdialog.cpp]: I prefer separation of UI from logic - so is it possible to have some external "importer" itself, which only performs some actions when Dialog is accepted (so no logic except for display one is in dialog itself)

File: src/component-importer/importcomponentdialog.cpp (hunk 3 above)
Comment (hcorg) [reply src/component-importer/importcomponentdialog.cpp]: ok, add some `// TODO` here so we'll find it during some clean up :)

File: asn1acn.pro (hunk 1 above)
Comment (psockoN7) [reply asn1acn.pro]: Added it to `libraries` , removed subfolder. I did not know what the `libraries` for at first, but I think new files fit there.

File: src/asn1acn.cpp (hunk 2 above)
Comment (psockoN7) [reply src/asn1acn.cpp]: Splitted the function into smaller chunks.

File: src/component-importer/importcomponentdialog.cpp (hunk 4 above)
Comment (psockoN7) [reply src/component-importer/importcomponentdialog.cpp]: I agree, Importer class was added. I believe that now logic is sufficiently separated from UI.

**Pull request merged**